joblib/hashing.py

```diff
@@ -86,7 +86,11 @@ def save(self, obj):
             else:
                 cls = obj.__self__.__class__
                 obj = _MyHash(func_name, inst, cls)
-        Pickler.save(self, obj)
+
+        if obj == type(None):
+            Pickler.save_none(self, obj)
```

Comment:
From the code above, I think it might be clearer to use the same strategy as in l.83, saying `obj = _MyHash('typenone', None)` in a `elif`?